scripts/update/main.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/google/go-github/v33/github"
+)
+
+type Repository struct {
+	Repo     string `csv:"repo"`
+	Metadata string `csv:"metadata,omitempty"`
+}
+
+// Programmatically gets Envoy's dependencies and add to projects.
+func EnvoyDeps() string {
+	client := github.NewClient(nil)
+	ctx := context.Background()
+	fo, _, _, err := client.Repositories.GetContents(ctx, "envoyproxy", "envoy", "bazel/repository_locations.bzl", nil)
```

Comment:
Can this be parameterized? I am not aware of the Bazel format being consistent across projects? If it then this can be requested for other bazel projects?

Reply:
The dictionary with project_url key structure is Envoy-specific.

Bazel projects will pull in deps in specific files, and you could grep through them for things matching the github.com/{}/{} pattern.

We could do that here and remove the dupes we have through the project URL and dep URL.

Reply:
e.g. a typical bazel dep file https://github.com/grpc/grpc/blob/e9de13e6ad98327aa6fcb017ebf375f870a3134b/bazel/grpc_deps.bzl#L157

Reply:
> The dictionary with project_url key structure is Envoy-specific so we can parse it as a python dict and do analysis/create tables with scripts.
> 
> Bazel projects will pull in deps in specific files, and you could grep through them for things matching the github.com/{}/{} pattern.
> 
> We could do that here and remove the dupes we have through the project URL and dep URL.

Sounds good.

Reply:
@naveensrinivasan sorry one more question :) 
any other bazel project that might be interesting to add right now (for testing)? 
i was pointed to https://github.com/google/starlark-go which can interpret *.bzl files and using that API would make this parametrizeable on filename by checking for url values.

Reply:
> @naveensrinivasan sorry one more question :)

Not an issue 👍  

> any other bazel project that might be interesting to add right now (for testing)?

How about the grpc that you had referred? Could we try that?

> i was pointed to https://github.com/google/starlark-go which can interpret *.bzl files and using that API would make this parametrizeable on filename by checking for url values.

Reply:
All done! I added a TODO to change over to starlark, but for now I'm searching the .bzl files for the github string. using startlark-go was a little complicated because navigating the modules was different.

BTW. I joined metadata like
`project=grpc,project=envoy,dependecy=true`. Not sure if that was ideal.

Reply:
> All done! I added a TODO to change over to starlark, but for now I'm searching the .bzl files for the github string. using startlark-go was a little complicated because navigating the modules was different.
> 

Sounds good!

> BTW. I joined metadata like
> `project=grpc,project=envoy,dependecy=true`. Not sure if that was ideal.

This is great! Thanks!